test/e2e/tests/test_table.py

```diff
@@ -825,6 +825,53 @@ def test_create_gsi_pay_per_request(self, table_basic_pay_per_request):
             gsi,
         ]
 
+        # Patch k8s resource
+        k8s.patch_custom_resource(ref, cr)
+        k8s.wait_resource_consumed_by_controller(ref)
+        table.wait_until(
+            table_name,
+            table.gsi_matches(
+                [
+                    gsi
+                ],
+            ),
+            timeout_seconds=MODIFY_WAIT_AFTER_SECONDS*40,
+            interval_seconds=15,
+        )
+    def test_create_gsi_same_attributes(self, table_basic):
+        (ref, res) = table_basic
+
+        table_name = res["spec"]["tableName"]
+
+        # Check DynamoDB Table exists
+        assert self.table_exists(table_name)
+
+        # Get CR latest revision
+        cr = k8s.wait_resource_consumed_by_controller(ref)
+
+        # Creating two GSI using the same attributes
+        gsi = {
+            "indexName": "total-bill",
+            "keySchema": [
+                {
+                    "attributeName": "Total",
+                    "keyType": "HASH",
+                },
+                {
+                    "attributeName": "Bill",
+                    "keyType": "RANGE",
+                }
+            ],
+            "projection": {
+                "projectionType": "ALL",
+            }
+        }
```

Comment:
the controller is not able to patch this gsi and logs the error `ValidationException: One or more parameter values were invalid: Both ReadCapacityUnits and WriteCapacityUnits must be specified for index: total-bill`

Reply:
Weird, since It's n on-demand table It should not require the Read and Write capacity to be set, It's the bug that was solved by Julian [here](https://github.com/aws-controllers-k8s/dynamodb-controller/pull/85). But let me check.

Reply:
Can you review again? @a-hilaly

Reply:
Looks good to me and tests are passing, but still wondering why do we need to use a new table here? /cc @Julian-Chu can you please take a look?

Reply:
@a-hilaly  do you mean `table_basic_gsi_same_attr` is not needed or combine the new test case into existing test case?

Reply:
It's ok to use a seperate test function here. I'm more wondering why this [commit was needed](https://github.com/aws-controllers-k8s/dynamodb-controller/pull/88/commits/ead35f13a1ddb37d960ddd582fefe3fafe9ea081) - it's basically using a completely fresh table.

Reply:
>the controller is not able to patch this gsi and logs the error `ValidationException: One or more parameter values were invalid: Both ReadCapacityUnits and WriteCapacityUnits must be specified for index: total-bill`

Also do we know why did we get this error when adding this GSI to a that already a few other GSIs?

Reply:
because the billing mode of `table_basic` is changed by `test_update_billing_mode`, but the test case written by @joaquinservetto  is using on-demand mode.

Reply:
and I don't see any GSI in `table_basic`.

Reply:
> because the billing mode of table_basic is changed by test_update_billing_mode, but the test case written by @joaquinservetto is still using on-demand mode.

Makes sense. Thanks @Julian-Chu for helping my understand this :)

Thanks a lot for your contributions @joaquinservetto !

/approve
/lgtm

Reply:
Not a fixture, also not very expert on pytests stuff. I would assume that a `scope="class"` would make more sense here ?

/lgtm cancel

Reply:
Thank you guys, I'm not an expert either, should I modify the scope then?

Reply:
Maybe `class` or just `function` as the scope make more sense here.

Reply:
with`scope="class"`,  pytest will cache the fixture for entire class, so if we want `table_basic` to return different table name for different methods in a class,   like @joaquinservetto  mentioned,   I think `function` is more suitable.  

ref: https://docs.pytest.org/en/6.2.x/fixture.html#scope-sharing-fixtures-across-classes-modules-packages-or-session

Reply:
Could you take a look again? @Julian-Chu @a-hilaly

Reply:
LGTM,  my original idea is to reuse `table_basic` with function scope.  it can be refactored later.